### ISSUE TITLE
Unexpeted type tag 13996 found.

### ISSUE THREAD

**Issue by Javieraeros** [OneSignal/OneSignal-Gradle-Plugin#15]
When adding the gradle plugin I´m getting this error (the tag number changes every time)
Any idea?
### Project Setup
`build.gradle`
```Gradle
apply from: 'buildsystem/ci.gradle'
apply from: 'buildsystem/dependencies.gradle'

buildscript {
    repositories {
        jcenter()
        mavenCentral()
        google()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
        classpath 'com.jakewharton:butterknife-gradle-plugin:8.4.0'
        classpath 'me.tatarka:gradle-retrolambda:3.6.0'
        classpath "io.realm:realm-gradle-plugin:3.5.0"
        classpath 'com.google.gms:google-services:3.1.2'
    }
}
allprojects {
    repositories {
        maven {
            url "https://jitpack.io"
        }
        maven {
            url "https://maven.google.com"
        }
        google()
    }
    ext {
        androidApplicationId = 'com.iagt.android.opeople.presentation'
        androidVersionCode = 1
        androidVersionName = "1.0"
        testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
        testApplicationId = 'com.innopro.android.sample.presentation.test'
    }
}

task wrapper(type: Wrapper) {
    description 'Creates the gradle wrapper.'
    gradleVersion '4.1'
}

task runDomainUnitTests(dependsOn: [':domain:test']) {
    description 'Run unit tests for the domain layer.'
}

task runDataUnitTests(dependsOn: [':data:cleanTestDebugUnitTest', ':data:testDebugUnitTest']) {
    description 'Run unit tests for the data layer.'
}

task runUnitTests(dependsOn: ['runDomainUnitTests', 'runDataUnitTests']) {
    description 'Run unit tests for both domain and data layers.'
}

task runAcceptanceTests(dependsOn: [':presentation:connectedAndroidTest']) {
    description 'Run application acceptance tests.'
}
```

`presentation/build.gradle`
```Gradle
plugins {
    id 'com.onesignal.androidsdk.onesignal-gradle-plugin' version '0.8.0'
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'

repositories {
    maven { url 'https://maven.google.com' }
}
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.android.application'
apply plugin: 'com.jakewharton.butterknife'
configurations {
    iagtprodDebugCompile
}
android {
    def globalConfiguration = rootProject.extensions.getByName("ext")

    compileSdkVersion globalConfiguration.getAt("androidCompileSdkVersion")
    buildToolsVersion globalConfiguration.getAt("androidBuildToolsVersion")

    defaultConfig {
        minSdkVersion globalConfiguration.getAt("androidMinSdkVersion")
        targetSdkVersion globalConfiguration.getAt("androidTargetSdkVersion")

        applicationId globalConfiguration.getAt("androidApplicationId")
        versionCode globalConfiguration.getAt("androidVersionCode")
        versionName globalConfiguration.getAt("androidVersionName")
        testInstrumentationRunner globalConfiguration.getAt("testInstrumentationRunner")
        testApplicationId globalConfiguration.getAt("testApplicationId")
        multiDexEnabled true
        manifestPlaceholders = [onesignal_app_id: "XXXXXXXXXXXXXXXXXXXXXXXX",
                                onesignal_google_project_number: "REMOTE"]

    }

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    packagingOptions {
        exclude 'LICENSE.txt'
        exclude 'META-INF/DEPENDENCIES'
        exclude 'META-INF/ASL2.0'
        exclude 'META-INF/NOTICE'
        exclude 'META-INF/LICENSE'
        exclude 'META-INF/rxjava.properties'
    }

    lintOptions {
        quiet true
        abortOnError false
        ignoreWarnings true
        disable 'InvalidPackage'            //Some libraries have issues with this.
        disable 'OldTargetApi'
        //Lint gives this warning but SDK 20 would be Android L Beta.
        disable 'IconDensities'             //For testing purpose. This is safe to remove.
        disable 'IconMissingDensityFolder'  //For testing purpose. This is safe to remove.
    }

    signingConfigs {
        debug {
            storeFile file('../buildsystem/debug.keystore')
            storePassword 'XXXXXX'
            keyAlias 'XXXXXX'
            keyPassword 'XXXXX'
        }
    }

    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
    }
    flavorDimensions "default"
    productFlavors {
        iagtprod {
            dimension "default"
            buildConfigField "boolean","prodVersion", "false"
            resValue "string", "facebook_app_id","\"XXXXXXXXXXXXXXX\""
            resValue "string", "facebook_login_protocol_scheme","\"fbXXXXXXXXXXXX\""
        }
        prod{
            dimension "default"
            buildConfigField "boolean","prodVersion", "true"
            resValue "string", "facebook_app_id","\"XXXXXXXXXXXXXXX\""
            resValue "string", "facebook_login_protocol_scheme","\"fbXXXXXXXXXXXX\""
        }
    }
}

dependencies {
    def presentationDependencies = rootProject.ext.presentationDependencies
    def presentationTestDependencies = rootProject.ext.presentationTestDependencies

    annotationProcessor presentationDependencies.daggerCompiler
    annotationProcessor presentationDependencies.butterKnifeCompiler
    annotationProcessor presentationDependencies.fragmentArgsPro
    compile presentationDependencies.googleMap
    compile presentationDependencies.googleMapUtils
    compile presentationDependencies.googleLocation
    compile presentationDependencies.googlePlaces
    compile presentationDependencies.googleAuth
    compile presentationDependencies.firebaseCore
    compile presentationDependencies.firebaseAuth
    compile presentationDependencies.dagger
    compile presentationDependencies.butterKnife
    compile presentationDependencies.recyclerView
    compile presentationDependencies.rxJava
    compile presentationDependencies.rxAndroid
    compile presentationDependencies.picasso
    compile presentationDependencies.appCompat
    compile presentationDependencies.supportDesign
    compile presentationDependencies.transformations
    compile presentationDependencies.androidAnnotations
    compile presentationDependencies.fragmentArgs
    compile presentationDependencies.hideKeyboard
    compile presentationDependencies.joda
    compile presentationDependencies.pulsator
    compile presentationDependencies.calligraphy
    compile presentationDependencies.passwordedEdittext
    compile presentationDependencies.stickyList
    compile presentationDependencies.multiCollapsing
    compile presentationDependencies.facebook
    compile presentationDependencies.multiDex


    implementation project(path: ':domain')
    implementation project(path: ':data')
    /*prodCompile project(path: ':domain')
    prodCompile project(path: ':data')*/
/*
  androidTestCompile presentationTestDependencies.mockito
  androidTestCompile presentationTestDependencies.espresso
  androidTestCompile presentationTestDependencies.androidAnnotations*/

    compile 'fr.avianey.com.viewpagerindicator:library:2.4.1.1@aar'
    compile 'com.android.support:support-v4:25.4.0'
    compile 'com.android.support.constraint:constraint-layout:1.0.2'
    compile 'com.onesignal:OneSignal:3.7.1'
}
```

### Full Error
```Java
Error:Unexpected type tag 13996 found.
```

### Dependency Tree
```
FAILURE: Build failed with an exception.

* What went wrong:
Project 'app' not found in root project 'opeople'.

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.execution.taskpath.ProjectFinderByTaskPath$ProjectLookupException: Project 'app' not found in root project 'opeople'.
        at org.gradle.execution.taskpath.ProjectFinderByTaskPath.findProject(ProjectFinderByTaskPath.java:47)
        at org.gradle.execution.taskpath.TaskPathResolver.resolvePath(TaskPathResolver.java:49)
        at org.gradle.execution.TaskSelector.getSelection(TaskSelector.java:95)
        at org.gradle.execution.TaskSelector.getSelection(TaskSelector.java:81)
        at org.gradle.execution.commandline.CommandLineTaskParser.parseTasks(CommandLineTaskParser.java:42)
        at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:44)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$000(DefaultBuildConfigurationActionExecuter.java:25)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.proceed(DefaultBuildConfigurationActionExecuter.java:54)
        at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:44)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$000(DefaultBuildConfigurationActionExecuter.java:25)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.proceed(DefaultBuildConfigurationActionExecuter.java:54)
        at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:47)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
        at org.gradle.execution.DefaultBuildConfigurationActionExecuter.select(DefaultBuildConfigurationActionExecuter.java:36)
        at org.gradle.initialization.DefaultGradleLauncher$CalculateTaskGraph.run(DefaultGradleLauncher.java:268)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.initialization.DefaultGradleLauncher.constructTaskGraph(DefaultGradleLauncher.java:175)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:130)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:109)
        at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:78)
        at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:75)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:152)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:100)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:75)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$1.run(RunAsBuildOperationBuildActionRunner.java:43)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:40)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:51)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:47)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:30)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:39)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:25)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:80)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:53)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:57)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:64)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:59)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:44)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:45)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:30)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)

```

**Comment by laudarch** [comment]
Am getting tag errors too.

**Comment by smeurer** [comment]
`multiDexEnabled true` seems to be the cause. My simple TestApp runs without errors without multidex, but if I add the line, the tag error occurs. Though this is the cause, it is no solution, as nearly all apps use multidex.

**Comment by zulfuadar** [comment]
Have you fix it? I got same error after adding "multiDexEnabled true".
#  My app level modul gradle 

plugins {
    id 'com.onesignal.androidsdk.onesignal-gradle-plugin' version '0.8.0'
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
apply plugin: 'com.android.application'
apply plugin: 'io.fabric'

repositories {
    maven { url 'https://maven.google.com' }
    maven { url 'https://maven.fabric.io/public' }
}
apply plugin: 'com.android.application'

android {
    dexOptions {
        jumboMode true

    }
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion
    defaultConfig {
        manifestPlaceholders = [onesignal_app_id               : "2fe20af9-bdcf-49cd-bbfe-6ffd8f11f8c2",
                                // Project number pulled from dashboard, local value is ignored.
                                onesignal_google_project_number: "REMOTE"]
        applicationId 'com.engly.learnenglish.app'
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode rootProject.ext.versionCode
        versionName rootProject.ext.versionName
        vectorDrawables {
            useSupportLibrary = true
        }
        generatedDensities = []
        multiDexEnabled true


    }
    lintOptions
            {
                checkReleaseBuilds false
                abortOnError false

            }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
    aaptOptions {
        additionalParameters "--no-version-vectors"
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
    productFlavors {
    }
    buildToolsVersion '26.0.2'
}

dependencies {
    compile 'com.android.support:multidex:1.0.1'
    compile 'com.android.support:multidex:1.0.2'
    implementation 'com.google.firebase:firebase-messaging:11.8.0'
    compile fileTree(include: ['*.jar'], dir: 'libs')
    //support lib
    compile rootProject.ext.supportLibAppCompat
    compile rootProject.ext.supportLibPercentLayout
    compile rootProject.ext.supportLibRecyclerView
    compile rootProject.ext.supportLibCardView
    compile rootProject.ext.supportLibDesign
    //butterknife
    compile rootProject.ext.butterknife
    compile 'com.android.support:support-v4:27.0.2'
    annotationProcessor rootProject.ext.butterknifeApt
    //commons
    compile rootProject.ext.commonsIo
    compile rootProject.ext.appacheCommons
    //gson
    compile rootProject.ext.googleGson
    //circle indicator
    compile rootProject.ext.relexCircleIndicator
    //piccasso
    compile rootProject.ext.picasso
    //retrofit
    compile rootProject.ext.okhttp
    compile rootProject.ext.retrofit2
    compile rootProject.ext.retrofit2Converter
    //recyclerview tablayout
    compile rootProject.ext.recyclerTabLayout
    //viewpager animation
    compile rootProject.ext.viewpagerTransforms
    //page loader
    compile rootProject.ext.pageLoader
    //top position snackbar
    compile rootProject.ext.topSnackbar
    compile 'android.arch.persistence.room:runtime:' + rootProject.archRoomVersion;
    annotationProcessor 'android.arch.persistence.room:compiler:' + rootProject.archRoomVersion;
    implementation 'com.facebook.android:facebook-login:4.29.0'
    compile 'com.android.support:appcompat-v7:27.0.2'
    compile 'com.onesignal:OneSignal:[3.6.0, 3.99.99]'
    compile 'uk.co.chrisjenx:calligraphy:2.3.0'
    compile 'com.github.michaelye.easydialog:easydialog:1.4'
    compile 'com.hbb20:ccp:1.7.8'
    compile 'com.googlecode.libphonenumber:libphonenumber:8.4.2'
    compile 'com.google.android.exoplayer:exoplayer:r2.4.1'
    compile 'com.mikhaellopez:circularprogressbar:1.1.1'
    compile 'com.google.firebase:firebase-core:11.8.0'
    compile 'com.victor:lib:1.0.4'
    compile('com.crashlytics.sdk.android:crashlytics:2.7.1@aar') {
        transitive = true
    }
    compile 'me.grantland:autofittextview:0.2.1'
    compile 'com.appsee:appsee-android:2.3.3'
    compile 'com.android.support.constraint:constraint-layout:1.1.0-beta4'
    testCompile 'junit:junit:4.12'
}
apply plugin: 'com.google.gms.google-services'

**Comment by janbolat** [comment]
Any solution?

**Comment by jkasten2** [comment]
@smeurer Thanks for reporting the root setting causing the issue. I was able to reproduce the issue in the a unit test.

@janbolat @zulfuadar @Javieraeros Created a fix in the commit above and released version 0.8.1 with this fix included.

**Comment by zulfuadar** [comment]
@jkasten2 yeaaahh. It is fixed. Thanks..
